go.mod

```diff
@@ -1,6 +1,8 @@
 module github.com/pulumi/pulumi-yaml
 
-go 1.19
+go 1.21
+
+toolchain go1.22.3
```

Comment:
`go mod tidy` changed these, not sure if I should discard these changes :thinking:

Reply:
Fails because 
```
github.com/pulumi/esc@v0.9.2-0.20240910221656-328d3204100f requires go@1.21, but 1.19 is requested
```
Do we need to bump Go requirement here to v1.21?

Reply:
Yes